streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskExecutor.java

```diff
@@ -275,7 +275,7 @@ private void commitSuccessfullyProcessedTasks() {
     int punctuate() {
         int punctuated = 0;
 
-        for (final Task task : tasks.activeTasks()) {
+        for (final Task task : tasks.notPausedTasks()) {
```

Comment:
Also here unit tests would be great and easily doable.

Reply:
@cadonna Ok, I tried to add some unit tests for `TaskExecutor` and I think I was needing to mock quite a few things.  Does that seem right?  Or is there an easier way?

Reply:
Note to self:  Use `notPausedActiveTasks`.